### PR TITLE
Finish Issue 2.2 - Task 2.2

### PR DESCRIPTION
**Pull request by BrianArz**
Completed task as requested.

### CODE REVIEW

File: Task2_2/DiagonalMatrix.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                    }
+
+                    output += "\t";
+                }
+
+                output += "\n";
+            }
+            return output;
+        }
+
```
Comment (IgorKheidorov) [inline Task2_2/DiagonalMatrix.cs:90]: This is not an extension method for diagonal matrix (I will explain on next  lecture)

File: Task2_2/DiagonalMatrix.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        {
+            var output = "";
+            for (var index = 0; index < Size; index++)
+            {
+                for (var subindex = 0; subindex < Size; subindex++)
+                {
+                    if (index == subindex)
+                    {
+                        output += _mainDiagonal[index];
+                    }
```
Comment (IgorKheidorov) [inline Task2_2/DiagonalMatrix.cs:77]: 1. Simplify, use indexer -> output += _mainDiagonal[index, subindex]; // no need if.. else statement in this case
2.  index, subindex are not clear names, better rowIndex, columnIndex

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Cool, some changes are required

File: Task2_2/DiagonalMatrix.cs (hunk 1 above)
Comment (BrianArz) [reply Task2_2/DiagonalMatrix.cs:90]: Good Morning @IgorKheidorov 

Do you want me to change the method? Or I wait until next lecture? 

Thanks!

File: Task2_2/DiagonalMatrix.cs (hunk 2 above)
Comment (BrianArz) [reply Task2_2/DiagonalMatrix.cs]: Changes implemented.

File: Task2_2/DiagonalMatrix.cs (hunk 1 above)
Comment (IgorKheidorov) [reply Task2_2/DiagonalMatrix.cs:82]: Implement extension method ))) Now you know how to)

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]

File: Task2_2/DiagonalMatrix.cs (hunk 1 above)
Comment (BrianArz) [reply Task2_2/DiagonalMatrix.cs]: Hahahaha yes, last lectuture made it all clear. 

Changes implemented. Waiting for review

**Review by IgorKheidorov — APPROVED** [review]

**Pull request merged**